core/src/main/scala/org/bykn/bosatsu/SourceConverter.scala

```diff
@@ -402,11 +419,25 @@ final class SourceConverter(
     }
   }
 
+  private def assertResult[A](a: Result[A]): A =
+    a match {
+      case Ior.Right(a) => a
+      case Ior.Both(err, a) =>
+        // TODO: push result all the way to the top
+        throw err.head
+      case Ior.Left(err) =>
+        // TODO: push result all the way to the top
+        throw err.head
+    }
+
+  private def unTuplePattern(pat: Pattern.Parsed, region: Region): Pattern[(PackageName, Constructor), rankn.Type] =
```

Comment:
this is big enough for now, we can follow up removing this and only using the Result type.